server/routes/charts.js

```diff
@@ -1,32 +1,68 @@
 import { Router } from 'express';
 
 import ensureAuthenticated from '../utils/passport/ensure-authenticated';
+import { collections } from '../utils/mongoCollections'
+import { userProps } from '../utils/pagePropsUtil';
 
 import chartsListPage from '../templates/Chart.template'
 import newChartPage from '../templates/NewChart.template'
 
 const router = Router();
 
 router.route('/charts')
-  .get(ensureAuthenticated, async (_, res) => {
+  .get(ensureAuthenticated, async (req, res) => {
     try {
       return res.status(200).send(chartsListPage())
-    } catch (err) {
-      console.error(err.message)
+    } catch (error) {
+      console.error(error.message)
 
       return res.status(500).send({ message: err.message })
     }
   });
   
 router.route('/charts/new')
-  .get(ensureAuthenticated, async (_, res) => {
+  .get(ensureAuthenticated, async (req, res) => {
     try {
-      return res.status(200).send(newChartPage())
+      const user = userProps(req)
+
+      return res.status(200).send(newChartPage(user))
     } catch (error) {
-      console.error(err.message)
+      console.error(error.message)
 
       return res.status(500).send({ message: err.message })
     }
 })
 
+router.route('/api/v1/charts')
+  .post(ensureAuthenticated, async (req, res) => {
+    try {
+      const additionalVariables = {}
+      const { fieldLabelValueMap, title, variable, assessment } = req.body
+      
+      if (fieldLabelValueMap.length) {
+        for (let index = 0; index < fieldLabelValueMap.length; index++) {
+          const field = fieldLabelValueMap[index];
+          additionalVariables[field.label] = field.value
+        }
+      }
+
+      const { dataDb } = req.app.locals
+      const { result } = await dataDb
+        .collection(collections.charts)
+        .insertOne({
+          title, 
+          variable, 
+          assessment, 
+          owner: req.user, 
+          ...additionalVariables 
```

Comment:
instead of spreading these variables, what do you think about having an array field of value/label (and eventually color) objects?

Reply:
[78b1245](https://github.com/AMP-SCZ/dpdash/pull/84/commits/78b12458361f85aeb0919c885be4ca8c8f6c047b)